### PR TITLE
fix!: make circuit relay listen on addresses like other transports

### PR DESCRIPTION
**Pull request by achingbrain**
Refactors the circuit relay transport to behave like other transports

1. Specify the addresses to listen on in the `addresses.listen` config
2. Remove the `discoverRelays` option
3. Remove the transport manager listen call to just listen on configured addresses
4. Emit the `listen` event when listen addresses change like other transports

This fixes the instances where we miss relay address updates and the corresponding `self:peer:update` event never fires.

BREAKING CHANGE: The `discoverRelays` option has been removed, instead add one or more instances of `"/p2p-circuit"` to the libp2p config under the `addresses.listen` key

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/transport-circuit-relay-v2/src/errors.ts [hunk 1]
```diff
@@ -17,3 +17,27 @@ export class DurationLimitError extends Error {
...
+ * An attempt to open a relayed connection over a relayed connection was made
+ */
+export class DoubleRelayError extends Error {
+  static name: string = 'DoubleRelayError'
+  name: string = 'DoubleRelayError'
+}
+
+/**
+ * An attempt to open a relayed connection over a relayed connection was made
+ */
```
Comment (SgtPooki) [inline packages/transport-circuit-relay-v2/src/errors.ts:39]: It doesn't seem like this is used anywhere, but this is the same description as `DoubleRelayError`

```suggestion
/**
 * An attempt to open a relayed connection when we already have `reservationConcurrency` relay reservations
 */
```

File: packages/integration-tests/test/circuit-relay.node.ts [hunk 2]
```diff
@@ -352,45 +326,32 @@ describe('circuit-relay', () => {
...
-            tcp(),
-            circuitRelayTransport({
-              discoverRelays: 1
-            })
-          ]
+          addresses: {
+            listen: [
+              '/p2p-circuit',
+              '/p2p-circuit',
+              '/p2p-circuit'
```
Comment (SgtPooki) [inline packages/integration-tests/test/circuit-relay.node.ts:333]: This feels more odd than `discoverRelays: 3`

**Review by SgtPooki — COMMENTED** [review]
just a quick peek

File: packages/integration-tests/test/circuit-relay.node.ts (hunk 2 above)
Comment (achingbrain) [reply packages/integration-tests/test/circuit-relay.node.ts:333]: Tbh there’s not a great reason to listen on more than one relay and a number of downsides to it - bigger identify messages, bigger provider records/peer-info objects.

Also given that the number of browser nodes can vastly outnumber publicly dialable server nodes it’s probably not something we should encourage anyway as a small number of relay clients could end up consuming all available relay slots on the network.

File: packages/transport-circuit-relay-v2/src/errors.ts (hunk 1 above)
Comment (achingbrain) [reply packages/transport-circuit-relay-v2/src/errors.ts]: It's thrown when the relay reservation queue is full.

**Pull request merged**